### PR TITLE
[ARM CPU] Migration Arm Compute Library to 23.08 version

### PR DESCRIPTION
**Pull request by allnes**

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
We can ignore `openvino-lin-arm64`, which is already fixed separately.

File: src/plugins/intel_cpu/thirdparty/ACLConfig.cmake [hunk 1]
```diff
@@ -335,8 +335,6 @@ elseif(NOT TARGET arm_compute::arm_compute)
         set(arm_compute_full_path "${arm_compute}")
     endif()
 
-    list(APPEND ARM_COMPUTE_OPTIONS experimental_fixed_format_kernels=True)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/thirdparty/ACLConfig.cmake:338]: Why it was deleted? Is it default value in ACL now?

**Comment by dmitry-gorokhov** [comment]
Need internal performance validation before merge

File: src/plugins/intel_cpu/thirdparty/ACLConfig.cmake (hunk 1 above)
Comment (allnes) [reply src/plugins/intel_cpu/thirdparty/ACLConfig.cmake:338]: This param was deleted and now build is failed with it

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**